File: post/leetcode/cpp/two-strings-are-anagrams.cpp
/*
@Copyright:LintCode
@Author:   justice_103
@Problem:  http://www.lintcode.com/problem/two-strings-are-anagrams
@Language: C++
@Datetime: 15-07-07 12:14
*/

class Solution {
public:
    /**
     * @param s: The first string
     * @param b: The second string
     * @return true or false
     */
    bool anagram(string s, string t) {
        // write your code here
        for( auto c : s )
            if( t.find(c) == string::npos )
                return false;
        return true;
    }
};

File: post/leetcode/python/two-strings-are-anagrams.py
# coding:utf-8
'''
@Copyright:LintCode
@Author:   justice_103
@Problem:  http://www.lintcode.com/problem/two-strings-are-anagrams
@Language: Python
@Datetime: 15-08-23 02:41
'''

class Solution:
    """
    @param s: The first string
    @param b: The second string
    @return true or false
    """
    def anagram(self, s, t):
        # write your code here
        
        if len(s)!=len(t):
            return False
        
        s=list(s)
        t=list(t)
        
        s.sort()
        t.sort()
                
        return s==t
